src/lib/currency/currency.ml

```diff
@@ -57,6 +57,9 @@ end = struct
 
   type t = Unsigned.t [@@deriving sexp, compare, hash]
 
+  (* can't be automatically derived *)
+  let dhall_type = Ppx_dhall_type.Dhall_type.Natural
```

Comment:
Can we change this to a `string`? It looks like it shouldn't make a difference to much else, so would be good to do it sooner rather than later.